src/test/test-conf-parser.c

```diff
@@ -274,6 +274,18 @@ static const char* const config_file[] = {
         "2\\\n"
         "3\n",
 
+        "[Section]\n"
+        "#hogehoge\\\n"      /* continuation is ignored in comment */
+        "setting1=1\\\n"     /* normal continuation */
+        "2\\\n"
+        "3\n",
+
+        "[Section]\n"
+        "setting1=1\\\n"     /* normal continuation */
+        "#hogehoge\\\n"      /* commented out line in continuation is ignored */
+        "2\\\n"
+        "3\n",
```

Comment:
This is the same example as the middle case in you `hoge.sh` bash example. And this patch implements different behaviour than bash :( This is exactly what I proposed back in the day, but the choice was made to follow bash. Dunno, I guess we can decide to break from bash, but then this needs to be documented.

If we follow bash semantics, this will be interpreted as:
```
[Section]
setting1=1
2 3
```

Reply:
To clarify: I think the bash behaviour is surprising and not very useful, and I think the behaviour that the patch currently implements is nice, but if we decide to do that, it needs to be a concious and documented choice.

Reply:
The bash behaviour makes perfect sense given that the bash comment syntax is specifically a comment-to-end-of-line, and not a whole line comment construct.
In bash, 
```
foo # bcd
```

The lexer consumes `# bcd` as a comment.

```
foo \
# bcd \
efg
```

The lexer sees the continuation after `foo`, starts lexing the next line, sees the `#` and the consumes until the end of the line.  The continuation is met with the empty line prior to the comment mark - no further continuation is requested (it's commented out).  `efg` is a separate statement.

Whilst I think providing continuation through a comment seems reasonable with a line oriented syntax, it is not the expected behaviour for systems professionals who are used to the fact that the shells generally allow mid-statement comments, and parse all comments as such.